clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/security/KubernetesV2Credentials.java

```diff
@@ -455,12 +457,13 @@ private void determineOmitKinds() {
     List<KubernetesKind> allKinds = KubernetesKind.getValues();
 
     log.info("Checking permissions on configured kinds for account {}... {}", accountName, allKinds);
-    Set<KubernetesKind> unreadableKinds = allKinds.parallelStream()
+    Map<KubernetesKind, String> unreadableKinds = allKinds.parallelStream()
       .filter(k -> k != KubernetesKind.NONE)
-      .filter(k -> !omitKinds.contains(k))
-      .filter(k -> !canReadKind(k, checkNamespace))
-      .collect(Collectors.toCollection(ConcurrentHashMap::newKeySet));
-    omitKinds.addAll(unreadableKinds);
+      .filter(k -> !omitKinds.keySet().contains(k))
+      .collect(Collectors.toConcurrentMap(k -> k, k -> checkIfKindError(k, checkNamespace))).entrySet().parallelStream()
```

Comment:
Should keep it as a `Stream` instead of calling `Collect` then immediately converting back to a `Stream`